### ISSUE TITLE
How to set priorityClassName for the pods/components of Calico?

### ISSUE THREAD

**Issue by tomsucho** [tigera/operator#2172]
## Expected Behavior
I was trying to use Calico for Network Security Policy and so it is important I think that all Calico components/pods get always scheduled fine. However in EKS nodes have limits of pods that can be placed on a particular instance type. I would like to set higher priority class for Calico pods that the operator creates so that they always get scheduled fine, especially for DaemonSets it creates.

## Current Behavior
Currently when I tried adding a `priorityClassName:` to the default Intallation cr it rejects it, so wondering how to pass this to the Operator.

## Possible Solution
Expose this via the Installtions CR just like I can see imagePullSecrets already there?

## Steps to Reproduce (for bugs)

NA

## Your Environment
EKS 1.21

* Operating System and version:
* Link to your project (optional):

**Comment by tmjd** [comment]
This is currently not available to configure.

What priority class are you hoping to use? We set the calico-node DaemonSet to `system-node-critical` which seems like the appropriate priorityClassName to use.

**Comment by tomsucho** [comment]
Indeed I can see that now, so this should work I think, but I'm sure the pod was stuck in Pending and events were showing max pods per node reached. In this case I would have expected the scheduler to evict any other pod with lower priority class and let the calico stuff to run on the node, but seem something went wrong. I'll keep an eye on this if it happens again I will dig deeper, but looks like you are doing the right thing already so closing this and thanks for prompt clarification of this!

**Comment by tomsucho** [comment]
actually re-opening this to make sure as I can see one of those stuck again:
```
calico-system                            csi-node-driver-v6zpv                                             0/2     Pending                           0          97s

kubectl -n calico-system get po csi-node-driver-v6zpv -o yaml | grep -i priority
  preemptionPolicy: PreemptLowerPriority
```
so it has low prio meaning not really need to run on the node and this Pending could be ignored?

**Comment by tmjd** [comment]
The `preemptionPolicy: PreemptLowerPriority` means the pod can preempt pods with a lower priority according to [the docs](https://kubernetes.io/docs/concepts/scheduling-eviction/pod-priority-preemption/#non-preempting-priority-class).

I think the issue is that it was missed that the CSI daemonset needed to be set to the `system-node-critical` priority class. 
The update adding the class to the CSI daemonset was merged in #2124 and has been picked to the v1.28 release branch (#2179) but there hasn't been a release created yet with that update.

**Comment by tomsucho** [comment]
okay that makes sense and looks like this is already covered in the future release, so closing this again then and thanks again for clarifications
